### PR TITLE
cleanup some util functions and enable back CLI binary validation

### PR DESCRIPTION
**Pull request by saidu941**
This PR includes
Summary of changes:
> Cleaned up commonly used util functions
> Enabled back CLI binary validation
> Added Unit tests for files and slices

### CODE REVIEW

File: pkg/string/slice.go [hunk 1]
```diff
@@ -61,3 +66,22 @@ func SliceAddString(slice []string, s string) ([]string, bool) {
 	}
 	return slice, false
 }
+
+// compareSlices compares 2 string slices after sorting
+func CompareTwoSlices(slice1 []string, slice2 []string) bool {
+	sort.Strings(slice1)
+	sort.Strings(slice2)
+	log := zap.S().With("test")
```
Comment (bhatpmk) [inline pkg/string/slice.go:74]: Do we really need to log anything from here ?

File: pkg/files/testdata/testdir1/dir1/cluster-file [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Copyright (c) 2022, Oracle and/or its affiliates.
```
Comment (bhatpmk) [inline pkg/files/testdata/testdir1/dir1/cluster-file:1]: Will you be able to test without adding all these empty files to repository ?

File: tests/e2e/verify-distribution/verifydistribution/verify_distribution_test.go [hunk 3]
```diff
@@ -38,14 +40,19 @@ var opensourcefileslistbydir = map[string][]string{
...
 
 var fullBundleFileslistbydir = map[string][]string{
-	"top": {"LICENSE", "README.md", "README.html", "bin", "images", "manifests"},
-	"bin": {"bom_utils.sh", "darwin-amd64", "darwin-arm64", "linux-amd64", "linux-arm64", "vz-registry-image-helper.sh"},
-	//"vz":        {"vz"},
-	"manifests": {"charts", "k8s", "profiles", "verrazzano-bom.json"},
-	"k8s":       {"verrazzano-platform-operator.yaml"},
+	"top":          {"LICENSE", "README.md", "README.html", "bin", "images", "manifests"},
+	"bin":          {"bom_utils.sh", "darwin-amd64", "darwin-arm64", "linux-amd64", "linux-arm64", "vz-registry-image-helper.sh"},
+	"darwin-amd64": {"vz"},
```
Comment (bhatpmk) [inline tests/e2e/verify-distribution/verifydistribution/verify_distribution_test.go:49]: If the test is checking whether the directory exists or not, can you just define one slice with the name "vz" and use ?

File: pkg/string/slice.go (hunk 1 above)
Comment (saidu941) [reply pkg/string/slice.go]: Addressed and updated PR

File: pkg/files/testdata/testdir1/dir1/cluster-file (hunk 2 above)
Comment (saidu941) [reply pkg/files/testdata/testdir1/dir1/cluster-file]: Addressed and updated PR

File: pkg/files/files_test.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+package files
+
+import (
+	asserts "github.com/stretchr/testify/assert"
+	"regexp"
+	"testing"
+)
+
+func TestGetMatchingFiles(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (saidu941) [reply pkg/files/files_test.go:16]: Resolved and updated PR

**Review by bhatpmk — APPROVED** [review]

**Pull request merged**